common/rule-processor/src/RuleHelper.js

```diff
@@ -24,7 +24,7 @@ class RuleMatcher {
       finalString = finalString.replace(new RegExp("[$]" + index, "g"), matchValue);
     });
 
-    return finalString;
+    return { destination: finalString, matches };
```

Comment:
This should be one level up. In `checkRegexMatch` and `checkWildcardMatch`. This method should only replace the corresponding match variables in the final destination